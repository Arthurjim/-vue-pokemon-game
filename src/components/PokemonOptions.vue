<template>
    <div class="options-container">
        <ul>
            <li v-for="pokemon in pokemons" :key="pokemon.id">
                <button
                    :disabled="disabledButtons()"
                    @click="$emit('selection', pokemon.id)"
                    :class="{'btn_disabled' : getStylesOptions(pokemon.id)}"
                >
                    <div v-if="getAnswer(pokemon.id)">
                        <img src="../assets/poke.png" alt="poke" />
                    </div>
                    {{ pokemon.name }}
                     <div v-if="getAnswer(pokemon.id)">
                        <img src="../assets/poke.png" alt="poke" />
                    </div>
                </button>
            </li>
        </ul>
    </div>
</template>

<script>
export default {
    data() {
        return {
            disabled: false,
        };
    },
    props: {
        pokemons: {
            type: Array,
            required: true,
        },
        pokemonSelected: {
            type: Number,
        },
        pokemonId: {
            type: Number,
        },
    },
    methods: {
        disabledButtons(){
            if(this.pokemonSelected){
                return true;
            }else{
                return false;
            }
        },
        // function para agregarle estilos a los botones no seleccionados
        getStylesOptions(id){
            if(this.pokemonSelected){
                if(this.pokemonSelected === id){
                    return false;
                }else{
                    return true;
                }
            }
        },
        getAnswer(id){
            if(this.pokemonSelected){
                if(this.pokemonId === id){
                    return true;
                }else{
                    return false;
                }
            }
        }
    },
    watch: {
        pokemonId(newId, oldId) {
           disabledButtons()
        },
        
    },
};
</script>

<style scoped>
ul {
    list-style-type: none;
}
li {
    background-color: rgb(3, 3, 216);
    background-color: rgba(0, 0, 0, 0.411);
    border-radius: 5px;
    border: 2px solid rgb(255, 238, 0);
    cursor: pointer;
    margin-bottom: 10px;
    height: 50px;
    width: 250px;
}

li:hover {
    background-color: rgba(0, 0, 0, 0.26);
}
li button {
    appearance: none;
    background: none;
    border: none;
    color: rgb(255, 238, 0);
    font-family: "Supermercado One", cursive;
    font-size: 16px;
    height: 100%;
    letter-spacing: 2px;
    text-shadow: 0px 2px 5px rgb(3, 3, 216);
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}
button div{
    display: inline-flex;
    justify-content: center;
    align-items: center;
    height: 50px;
    margin: 0 10px;
}
.btn_disabled {
    opacity: 0.3;
}

.options-container {
    display: flex;
    justify-content: center;
}
img{
    width: 20px;
    height: 20px;
    display: inline-flex;
}
</style>
