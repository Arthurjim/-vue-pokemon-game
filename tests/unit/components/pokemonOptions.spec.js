import { shallowMount } from "@vue/test-utils";
import PokemonOptions from '@/components/PokemonOptions';
import {pokemonsMock} from '../mocks/pokemons.mock'
describe('Pruebas en Componente', () => {
   
    const wrapper = shallowMount(PokemonOptions,{
        props:{
            pokemons:pokemonsMock
        }
    })
    test('Debe hacer match con el snaptshot', () => {
        expect(wrapper.html()).toMatchSnapshot()
     })

     test('Debe de mostra las 4 ocpciones correctamente', () => { 
        const liTags= wrapper.findAll('li')
        // const isTrue = liTags.map(item=>item.text().length > 0 ? true : false)
        // expect(isTrue.includes(false)).toBe(false)

        expect(liTags[0].text()).toBe('bulbasaur')
        expect(liTags[1].text()).toBe('ivysaur')
        expect(liTags[2].text()).toBe('venusaur')
        expect(liTags[3].text()).toBe('charmander')

      })

      test('Debe de emitir "selection" con sus respectivos parÃ¡metros al hacer click', () => { 
        const [p1,p2,p3,p4]= wrapper.findAll('li')
        const buttonA = p1.find('button')
        buttonA.trigger('click')
        expect(wrapper.emitted('selection').length).toBe(1)
          
       })
})