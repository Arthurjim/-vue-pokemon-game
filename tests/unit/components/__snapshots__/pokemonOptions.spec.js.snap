// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Componente Debe hacer match con el snaptshot 1`] = `
<div class="options-container">
  <ul>
    <li><button class="">
        <!--v-if--> bulbasaur
        <!--v-if-->
      </button></li>
    <li><button class="">
        <!--v-if--> ivysaur
        <!--v-if-->
      </button></li>
    <li><button class="">
        <!--v-if--> venusaur
        <!--v-if-->
      </button></li>
    <li><button class="">
        <!--v-if--> charmander
        <!--v-if-->
      </button></li>
  </ul>
</div>
`;
